UI/src/application.js

```diff
@@ -45,13 +45,15 @@ var URLRouter = function(map, backend, models) {
       var roadLinkReceived = backend.getRoadLinkByMMLId(mmlId);
       var layerSelected = eventbus.oncePromise('layer:speedLimit:shown');
       applicationModel.selectLayer('speedLimit');
-      var mapMoved = $.when(roadLinkReceived).then(function(response) {
-        var promise =  eventbus.oncePromise('layer:speedLimit:moved');
-        map.setCenter(new OpenLayers.LonLat(response.middlePoint.x, response.middlePoint.y), 12);
-        return promise;
-      });
-      $.when(layerSelected, mapMoved).then(function() {
-        eventbus.trigger('speedLimit:selectByMmlId', parseInt(mmlId, 10));
+      $.when(layerSelected).then(function() {
```

Comment:
It could. I did this refactoring and then ran into the fact that jQuery's promises are broken as hell.